### DISCUSSION TITLE
Post processing interpolated values

### DISCUSSION THREAD

**Discussion by kgominyuk** [i18next/i18next#1701] [Q&A]
Hello!

Could you please tell is there a way to allow post processing of translation result in interpolation?

I wrote post processor module and seems like it doesn't process interpolated values.

**Comment by jamuhl** [comment]
Postprocessing happens after nesting and interpolation https://github.com/i18next/i18next/blob/master/src/Translator.js#L351

edge cases might be using react-i18next Trans component

**Comment by kgominyuk** [comment]
Did you consider to allow somehow process nested values?

**Comment by jamuhl** [comment]
what you mean...i18next does all internal processing and as last thing passes the found result to any post processor.

**Comment by kgominyuk** [comment]
I implemented PostProcessorModule which append `context` if there is `count` in this way:
`context: options.count === 0 ? "zero" : options.count`

This is very useful and helps a lot. 

However this approach doesn't work with nested values. I could do something like this:
`"girlsAndBoys": "They have $t(girls, {\"count\": {{girls}},  \"context\": {{girls}} }) and $t(boys, {\"count\": {{boys}}, \"context\": {{boys}} })", 
`    
but nested expressions use prefix `_0` instead of `_zero` for `count === 0`
This is why I ask is there a way to use PostProcessorModule even for nested expressions.

BTW  "_implementations are encouraged to provide the ability to have special plural messages for 0 in particular, so that more natural language can be used_" (https://cldr.unicode.org/index/cldr-spec/plural-rules)

**Reply by jamuhl** [reply]
if you like you might try creating a pull request:

[loadPath: "/locales/{{lng}}/{{ns}}.{{some-random-hash}}.json"](https://github.com/i18next/i18next/blob/master/src/Translator.js#L445)

```js
if (needsPluralHandling) finalKeys.push((finalKey += pluralSuffix));
if (needsPluralHandling && options.count === 0 && (pluralSuffix === 'other' || pluralSuffix === 'one')) finalKeys.push((finalKey += 'zero'));
```
